src/Functions/getSetting.cpp

```diff
@@ -19,20 +19,23 @@ namespace ErrorCodes
 namespace
 {
 
+enum class ErrorHandling : uint8_t
+{
+    Exception,
+    Default,
+};
+
 /// Get the value of a setting.
 class FunctionGetSetting : public IFunction, WithContext
 {
 public:
-    static constexpr auto name = "getSetting";
-
-    static FunctionPtr create(ContextPtr context_) { return std::make_shared<FunctionGetSetting>(context_); }
-    explicit FunctionGetSetting(ContextPtr context_) : WithContext(context_) {}
+    explicit FunctionGetSetting(ContextPtr context_, String name_, ErrorHandling error_handling_) : WithContext(context_), name(name_), error_handling(error_handling_) {}
 
     String getName() const override { return name; }
     bool isDeterministic() const override { return false; }
     bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
-    size_t getNumberOfArguments() const override { return 1; }
-    ColumnNumbers getArgumentsThatAreAlwaysConstant() const override { return {0}; }
+    size_t getNumberOfArguments() const override { return ( name == "getSettingOrDefault" ? 2 : 1 ); }
```

Comment:
Relying on naming is not very good, rely on the mode instead

Reply:
Done, thanks!